# Data cleaning airports.csv

In [68]:
import pandas as pd

In [69]:
df = pd.read_csv('airports.csv')

In [70]:
df.shape

(7698, 14)

In [71]:
df.isna().sum()

AirportID                 0
Name                      0
City                     49
Country                   0
IATA                      0
ICAO                      0
Latitude                  0
Longitude                 0
Altitude                  0
Timezone                  0
DST                       0
Tz_database_time_zone     0
Type                      0
Source                    0
dtype: int64

- There are 49 empty city fields 

In [72]:
# Let's take a look
df[df['City'].isna()].head()

AirportID                                Name City       Country IATA  \
7031      11794  Minsk Mazowiecki Military Air Base  NaN        Poland   \N   
7032      11795            Powidz Military Air Base  NaN        Poland   \N   
7137      11900       King Salman Abdulaziz Airport  NaN  Saudi Arabia  DWD   
7138      11901                King Khaled Air Base  NaN  Saudi Arabia  KMX   
7158      11921                  Asahikawa Airfield  NaN         Japan   \N   

      ICAO   Latitude   Longitude  Altitude Timezone DST  \
7031  EPMM  52.195499   21.655899       604       \N  \N   
7032  EPPW  52.379398   17.853901       371       \N  \N   
7137  OEDM  24.449900   44.121201      3026       \N  \N   
7138  OEKM  18.297300   42.803501      6778       \N  \N   
7158  RJCA  43.794734  142.365432       377       \N  \N   

     Tz_database_time_zone     Type       Source  
7031                    \N  airport  OurAirports  
7032                    \N  airport  OurAirports  
7137                    \N  airport  OurAirports  
7138                    \N  airport  OurAirports  
7158                    \N  airport  OurAirports

- For our purposes we want IATA but looking at this it \N is not a proper IATA airport code

In [73]:
# Let's see how many \N IATA airport codes there are
df[df['IATA']=='\\N'].count()

AirportID                1626
Name                     1626
City                     1616
Country                  1626
IATA                     1626
ICAO                     1626
Latitude                 1626
Longitude                1626
Altitude                 1626
Timezone                 1626
DST                      1626
Tz_database_time_zone    1626
Type                     1626
Source                   1626
dtype: int64

- There are 1616 rows without a proper IATA airport code, so we'll get rid of these and with null value for City

In [74]:
df = df.dropna(subset = ['City'])

In [75]:
df.shape

(7649, 14)

- Perfect, (7698 - 7649 = 49) rows were removed

In [76]:
clean_df = df[df['IATA']!='\\N']

In [77]:
clean_df.shape

(6033, 14)

- Nice! (7649 - 1616 = 6033) are ready to go

In [78]:
# Only need Name, City, and IATA

In [79]:
clean_df = clean_df[['Name', 'City', 'IATA']]

In [80]:
clean_df.to_csv('clean_airports.csv', index=False)

In [81]:
test = pd.read_csv('clean_airports.csv')

In [82]:
test

Name                 City IATA
0                                  Goroka Airport               Goroka  GKA
1                                  Madang Airport               Madang  MAG
2                    Mount Hagen Kagamuga Airport          Mount Hagen  HGU
3                                  Nadzab Airport               Nadzab  LAE
4     Port Moresby Jacksons International Airport         Port Moresby  POM
...                                           ...                  ...  ...
6028                       Bilogai-Sugapa Airport  Sugapa-Papua Island  UGU
6029                                Ramon Airport                Eilat  ETM
6030                               Rustaq Airport          Al Masna'ah  MNH
6031                         Laguindingan Airport  Cagayan de Oro City  CGY
6032                  Desierto de Atacama Airport              Copiapo  CPO

[6033 rows x 3 columns]